newIDE/app/src/CommandPalette/CommandHooks.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import * as React from 'react';
+import { type Command } from './CommandManager';
+import CommandsContext from './CommandsContext';
+import PreferencesContext from '../MainFrame/Preferences/PreferencesContext';
+
+export const useCommand = (commandName: string, command: Command) => {
+  const commandManager = React.useContext(CommandsContext);
+  React.useEffect(
+    () => {
+      if (!command.enabled) return;
+      commandManager.registerCommand(commandName, command);
+      return () => commandManager.deregisterCommand(commandName);
+    },
+    // eslint-disable-next-line
```

Comment:
Is there a way to avoid this disabling?

Reply:
Right now, it's disabled because the `command` object will be a new object for each Mainframe render, so adding `command` to the dependency list would re-register commands on each Mainframe render.
Even if I destructure the properties of `command`, `command.displayText` is a new object on each Mainframe render, due to the `t` macro. So can't include that as dependency either :(

One possible solution is for the hook to just take a static `displayText` string, and apply the `t` macro inside the hook instead. Then destructuring should probably work. I'll try doing that (I have an exam tomorrow, so might take some time :sweat_smile: )